adept/container/actorlearner/learner_container.py

```diff
@@ -0,0 +1,374 @@
+# Copyright (C) 2018 Heron Systems, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import os
+from time import time
+import datetime
+
+import numpy as np
+import ray
+from ray.experimental.sgd.utils import find_free_port
+import torch
+import torch.distributed as dist
+from torch.utils.tensorboard import SummaryWriter
+
+from adept.container.base import Container
+from adept.network import ModularNetwork
+from adept.registry import REGISTRY
+from adept.container.actorlearner.rollout_queuer import RolloutQueuerAsync
+from adept.utils import dtensor_to_dev, listd_to_dlist
+from adept.utils.logging import SimpleModelSaver
+
+
+class NCCLOptimizer:
```

Comment:
move to a different file, in future we should have 'optimizer' package but for now can just leave in 'container' package under 'base/nccl_optimizer.py'